windows-msi/README.rst

```diff
@@ -0,0 +1,99 @@
+   on this.
+
+4. Open Command Prompt on Windows and ``cd`` to ``openvpn-build\windows-msi``
+   folder.
+
+   To transfer the openvpn-build site to a Windows computer, you can copy it,
+   or share it using Samba on the Linux box. On a Windows computer mount the
+   Samba share as a drive (e.g. ``Z:``), since you cannot ``cd`` to a UNC path
+   of form ``\\computer\share\path``.
+
+5. Run ``cscript build.wsf`` to build the packages. The ``build.wsf`` is a
+   simple Makefile type building tool developed to avoid Microsoft Visual
+   Studio or GNU Make requirements. Refer to ``build.wsf`` for exact usage::
+
+    C:\openvpn-build\windows-msi>cscript build.wsf /?
+    Microsoft (R) Windows Script Host Version 5.812
+    Copyright (C) Microsoft Corporation. All rights reserved.
+    
+    Packages OpenVPN for Windows.
+    Usage: build.wsf [<command>] [/a]
+    
+    Options:
+    
+    <command> : Command to execute (default: "all")
+    a         : Builds all targets even if output is newer than input
+    
+    Commands:
+    all     Builds MSI packages and EXE installer
+    msi     Builds MSI packages only
+    exe     Builds EXE installer only
+    clean   Cleans intermediate and output files
+
+6. The MSI packages and EXE installer will be put to ``image`` subfolder.
+
+
+Digital Signing
+~~~~~~~~~~~~~~~
+
+The ``build.wsf`` tool does not support digital signing of MSI and EXE files
+(yet). For official packages, please keep the following guidelines in mind:
+
+- Build MSI packages first: ``cscript build.wsf msi``. Sign them. Build EXE
+  installer next: ``cscript build.wsf exe``. Sign it. This ensures the MSI
+  packages inside EXE installer payload are signed.
+
+- The ``cscript build.wsf exe`` does not build MSI packages. This is a safety
+  feature to prevent accidental rebuild of already signed MSI files, should
+  something accidentally touch any of the MSI package source files.
+
+- EXE installer does not ask for elevation. It extracts and launches
+  appropriate MSI package unelevated. The UAC elevation is requested only
+  later when MSI package actually starts the install process. Therefore, it is
+  vital to digitally sign MSI packages. Digital signing of EXE installer is
+  optional, but recommended to decrease the chance Windows SmartScreen will
+  treat our EXE installer as malware on downloads.
+
+- When signing MSI packages, set a signature description (``/d`` flag with
+  ``signtool.exe`` utility). The ``msiexec.exe`` saves the MSI package under
+  some random name and launches an elevated process to install it. When the
+  signature on the MSI package contains no description, Windows displays the
+  MSI filename instead on the UAC prompt. Now MSI having a random filename,
+  the UAC prompt gets quite confusing. Therefore, we strongly encourage you to
+  set a description in the MSI signature accurately describing the package
+  content.
+
+- MSI packages do not support multiple signatures. Sign them only once using
```

Comment:
We (=OpenVPN project) do not officially support Windows Vista anymore. So SHA-256 signature should probably be the recommendation.

Reply:
True.

It makes sense to update the document then since it also mentions double signing of EXE/DLL binaries. If Vista is no longer supported, SHA-1+SHA-256 double signing is a thing of past.

Reply:
You're referring to documentation on Trac, right? I had a quick look and the signing part requires a major overhaul. I created an internal JIRA ticket assigned to myself to fix that after the next tap-windows6 release.

Reply:
Resolved in a9b1331c763d388aaa51b72d9e12deb5bcbfe0d9.